src/cow-react/modules/limitOrders/hooks/useParameterizeLimitOrdersInMenu.ts

```diff
@@ -0,0 +1,34 @@
+import { Routes } from '@cow/constants/routes'
+import { useWeb3React } from '@web3-react/core'
+import { useAtomValue } from 'jotai/utils'
+import { limitOrdersAtom } from '@cow/modules/limitOrders/state/limitOrdersAtom'
+import { useEffect } from 'react'
+import { useSetAtom } from 'jotai'
+import { mainMenuUrlOverridesAtom } from '@cow/modules/mainMenu/state/mainMenuUrlOverridesAtom'
+import { MainMenuItemId } from '@cow/modules/mainMenu/constants/mainMenu'
+
+export function useParameterizeLimitOrdersInMenu() {
+  const { chainId } = useWeb3React()
+  const { inputCurrencyId, outputCurrencyId } = useAtomValue(limitOrdersAtom)
+  const overrideMainMenu = useSetAtom(mainMenuUrlOverridesAtom)
+
+  useEffect(() => {
+    const route = parameterizeLimitOrdersRoute(chainId, inputCurrencyId, outputCurrencyId)
+
+    overrideMainMenu({ [MainMenuItemId.LIMIT_ORDERS]: route })
```

Comment:
And it would be weird to make `Header` dependent on `Limit-orders`. So, using `mainMenuUrlOverridesAtom` we avoid of `Dependency inversion` principle breach.